src/quo/components/slideshow/slider_bar/view.cljs

```diff
@@ -0,0 +1,162 @@
+
+(defn- item
+  [{:keys [index active-index total-length active? customization-color theme blur?]
+    :or   {customization-color :blue
+           theme               :light}}]
+  (let [shared-scale (reanimated/use-shared-value (get-scale index active-index total-length))]
+    (rn/use-mount (fn []
+                    (let [new-scale-value (get-scale index active-index total-length)]
+                      (reanimated/animate shared-scale new-scale-value))))
+    [rn/view
+     {:key   index
+      :style (style/item-wrapper {:size    ITEM_SIZE
+                                  :spacing ITEM_SPACING})}
+     [reanimated/view
+      {:accessibility-label (make-item-label index active?)
+       :style
+       (reanimated/apply-animations-to-style
+        {:transform [{:scale shared-scale}]}
+        (style/item {:size                ITEM_SIZE
+                     :spacing             ITEM_SPACING
+                     :active?             active?
+                     :customization-color customization-color
+                     :theme               theme
+                     :blur?               blur?}))}]]))
+
+(defn- get-item-layout
+  [_ index]
+  (let [length (+ ITEM_SIZE ITEM_SPACING)]
+    #js {:length length
+         :index  index
+         :offset (* index length)}))
+
+(defn- view-internal
+  [{:keys [total-amount active-index customization-color blur? accessibility-label]
+    :or   {customization-color :blue
+           blur?               false
+           active-index        0
+           total-amount        1}}]
+  (let [theme            (quo.theme/use-theme)
+        flat-list-ref    (rn/use-ref-atom nil)
+        on-flat-list-ref (rn/use-callback #(reset! flat-list-ref %) [flat-list-ref])
+        center-position  0.5
+        data             (range total-amount)
+        scroll-to-index  (rn/use-callback
+                          (fn []
+                            (.scrollToIndex ^js @flat-list-ref
+                                            #js {:animated     true
+                                                 :index        active-index
+                                                 :viewOffset   ITEM_SPACING
+                                                 :viewPosition center-position}))
+                          [flat-list-ref active-index])
+        render-fn        (rn/use-callback
+                          (fn [i]
+                            [item
+                             {:index               i
+                              :total-length        total-amount
+                              :active-index        active-index
+                              :active?             (= i active-index)
+                              :customization-color customization-color
+                              :theme               theme
+                              :blur?               blur?}])
+                          [total-amount active-index customization-color theme])]
```

Comment:
:thinking: 
I'm not sure if this is the best practice here. :thinking: 

`render-fn` is not  a callback, our wrapper will use this function to create a reagent component, so the best practice is to declare it as any other reagent component (i.e. in the global scope of this namespace) and, to avoid undesirded re-renders or performance problems, it should be a pure component, if we want to provide extra parameters the flat-list allows it by using the key `:extra-data` (as far as I remember).

I'm not sure about what will happen if we memoize this component with `rn/use-callback`.

CC: @ilmotta  any thoughts?
I think we should make our wrapper of `flat-list` closer to the react-native's API and more transparent in terms of abstractions.